lib/request-base.js

```diff
@@ -155,17 +155,16 @@ RequestBase.prototype._retry = function() {
 
   // node
   if (this.req) {
-    delete this.req;
+    this.req = null;
+    this.req = this.request();
   }
 
   this._aborted = false;
   this.timedout = false;
   delete this._timer;
```

Comment:
I don't understand why these fields are deleted here. The timers are probably cancelled when the error happens (they should…), and should be irrelevant later.